src/components/Sidebar/Sidebar.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  
+<aside 
+
+  class="w-56 absolute inset-y-0 left-0 transform -translate-x-full transition duration-200 ease-in-out  "
+  :class="{'relative -translate-x-0':showSideBar}"
+  aria-label="Sidebar">
+  <div class="overflow-y-auto py-4 px-3  bg-gray-50  h-screen dark:bg-gray-800">
+    <div class="flex items-center justify-between p-2 tracking-widest text-base font-bold text-gray-900 rounded-lg dark:text-white ">
+      <p class="p-[10px]">
+        BeforeClass
+      </p>
+    <button class="hover:bg-gray-100 dark:hover:bg-gray-700 p-2 rounded-lg" @click="showSideBar = !showSideBar">
+      <svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white " fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M11 19l-7-7 7-7m8 14l-7-7 7-7"></path></svg>
+    </button>
+    </div>
+      <ul class="space-y-2 p-[10px] mt-[15px]">
+        <li>
```

Comment:
These `<li>` should be a standalone component that takes in a string & icon(svg) as parameter and display it (reduce duplication for `Modules`, `Settings` and any other future tabs